plugins/WindowManager/TopLevelWindowModel.cpp

```diff
@@ -725,4 +731,12 @@ void TopLevelWindowModel::activateNullWindow()
 {
     if (!m_nullWindow->focused())
         m_nullWindow->activate();
+}
+
+void TopLevelWindowModel::closeAllWindows()
+{
+    m_closingAllApps = true;
+    for (auto win : m_windowModel) {
+        win.window->close();
+    }
 }
```

Comment:
Please add a newline at end of file.

Reply:
why do we need this? it makes it really inconvenience when vscode removes this automatically anyway...

Reply:
In Code, enable `Files: Insert Final Newline` and `Files: Trim Final Newlines`. This will ensure there is exactly one newline character at the end of each file.

As for why... https://stackoverflow.com/a/729795

Reply:
Thanks :) will update it :)